diofant/solvers/ode.py

```diff
@@ -3792,16 +3792,12 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
     numsyms = [next(numsyms) for i in range(n + 1)]
     serlist = []
     for ser in [p, q]:
-        # Order term not present
-        if ser.is_polynomial(x) and Poly(ser, x).degree() <= n:
-            if x0:
-                ser = ser.subs({x: x + x0})
-            dict_ = Poly(ser, x).as_dict()
-        # Order term present
-        else:
+        if not ser.is_polynomial(x) or Poly(ser, x).degree() > n:
             tseries = series(ser, x=x, x0=x0, n=n + 1)
-            # Removing order
-            dict_ = Poly(list(ordered(tseries.args))[: -1], x).as_dict()
```

Comment:
I also think, that this "removing" is a meaningless part of code.

This "works" in sympy only due to very low test coverage (one test, that I have found!).  In this example, the tseries is just a `4*x**4` and "removing order" give us just a `Poly(4, x)`.  Perhaps, there was assumed something like `Poly(tseries.removeO(), x)`?